### PR TITLE
added docker scan and updated formatting of other jobs/cmds

### PR DESCRIPTION
**Pull request by Cwagne17**
Completes [Story 8658](https://app.shortcut.com/clarkcan/story/8658/implement-job-scan-docker)

Updates some changes needed to the deploy-ecs PR, which will be merged into the PR once this is approved.

Also adds the scan-docker job, please be very critical of it there is a lot going on here. If we should break this into scripts with tests let me know.

Here is the slack notif.
Screen Shot 2022-04-15 at 12 20 39 PM

I tested some bucket permissions on my personal AWS account. I have tested uploading, just access. This is the bucket policy I came up with to only allow access to authenticated users in our account.
```json
{
    "Version": "2012-10-17",
    "Id": "Policy1650037755070",
    "Statement": [
        {
            "Sid": "Stmt1650037750571",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::<ACCOUNT_ID>/*"
            },
            "Action": [
                "s3:GetObject",
                "s3:GetObjectVersion"
            ],
            "Resource": "arn:aws:s3:::<BUCKET_NAME>/*"
        }
    ]
}
```

In terms of what needs to be done to other environments.
We will need to create a bucket to store these objects.
We will need to verify that the MACHINE_AWS_S3 user can push to this bucket
We need to consider adding a new private channel to send cd-docker-scans notifs to so that our circleci-deployments channel doesn't get bloated.

### CODE REVIEW

**Review by EVogel1999 — APPROVED** [review]

File: src/jobs/scan-docker.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+        docker scan --json \
+          --accept-license \
+          --severity high \
+          --file <<parameters.dockerfile-path>> \
+          <<parameters.image>>:<<parameters.tag>> > "/tmp/${OUTPUT_FILENAME}.json"
+
+        apk add jq
+
+        if jq -e '.[0].vulnerabilities == []' "cards-link-check:1.0.0.json" > /dev/null; then
```
Comment (StarChaser44) [inline src/jobs/scan-docker.yml:109]: Why is there cards-link-check here?

**Review by StarChaser44 — COMMENTED** [review]
One question :)

File: src/jobs/scan-docker.yml (hunk 1 above)
Comment (Cwagne17) [reply src/jobs/scan-docker.yml:109]: Oh good catch!

**Review by StarChaser44 — APPROVED** [review]

**Review by mfranz1 — APPROVED** [review]

File: src/jobs/scan-docker.yml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      (defaults to AWS_DEFAULT_REGION)
+  aws-s3-bucket:
+    type: env_var_name
+    default: AWS_S3_BUCKET
+    description: |
+      Env var of AWS s3 bucket to put object to
+      (defaults to AWS_S3_BUCKET)
+  docker-password:
+    type: env_var_name
+    default: DOCKER_PASSWORD
```
Comment (pzalep1) [inline src/jobs/scan-docker.yml:44]: Do these need to be the defaults? We use other variables right?

**Review by JosueMolinaMorales — APPROVED** [review]

**Pull request merged**